libpromises/fncall.c

```diff
@@ -271,19 +271,19 @@ static FnCallResult CallFunction(EvalContext *ctx, const Policy *policy, const F
 
     if (argnum != RlistLen(expargs) && !(fncall_type->options & FNCALL_OPTION_VARARG))
     {
-        fprintf(stderr, "Argument template mismatch handling function %s(", fp->name);
+        Log(LOG_LEVEL_ERR, "Argument template mismatch handling function %s(", fp->name);
```

Comment:
This won't work, Log() always prints whole lines. Try compositing the string and printing it in the end.

Reply:
Good catch, fixed.